fibonacci/main.go

```diff
@@ -3,12 +3,29 @@ package main
 import "fmt"
 
 func main() {
-	fmt.Println(fibonacci(7))
+	const n = 5
+	fmt.Println("Calculating %d sequence\n", n)
 }
 
-func fibonacci(n int) int {
-	// Fill me in!
-	// When n is 0 or 1, fibonacci(n) should just be n.
-	// When n is any larger number,
-	// fibonacci(n) should be the sum of the previous two fibonacci numbers.
+func fibonacci(n int, old, new, calc string) int {
+
+	if n == 0 {
+		return
+	}
+
+	fibonacci(n-1, old, calc, new)
+
+	fmt.Printf("  n  ", old, new)
+
+	fibonacci(n-1, calc, new, old)
```

Comment:
Here again, you're being guided by the Towers of Hanoi example and making things more complicated than they need to be.

One important difference is that `hanoi` printed out its results as it went along, rather than returning a value. This function must return a value: namely,

- 0 when `n` is 0
- 1 when `n` is 1
- the sum of the previous two fibonacci numbers otherwise.